lib/find.ts

```diff
@@ -1,36 +1,45 @@
 import * as fs from 'fs';
 import { join } from 'path';
 import { promisify } from 'util';
+import { MAX_SUPPORTED_FILE_SIZE } from './common';
 import { debug } from './debug';
 
 const readdir = promisify(fs.readdir);
-const stat = promisify(fs.stat);
+export const stat = promisify(fs.stat);
 
 export async function find(dir: string): Promise<string[]> {
-  const result: string[] = [];
-
   const dirStat = await stat(dir);
   if (!dirStat.isDirectory()) {
     throw new Error(`${dir} is not a directory`);
   }
 
   const paths = await readdir(dir);
-
+  const absolutePaths: string[] = [];
   for (const relativePath of paths) {
-    const absolutePath = join(dir, relativePath);
+    absolutePaths.push(join(dir, relativePath));
+  }
+
+  return await extractFilePaths(absolutePaths);
+}
+
+export async function extractFilePaths(
+  absolutePaths: string[],
+): Promise<string[]> {
+  const result: string[] = [];
+  for (const absolutePath of absolutePaths) {
     try {
-      const fileStat = await stat(absolutePath);
-      if (fileStat.isDirectory()) {
+      const fileStats = await stat(absolutePath);
```

Comment:
should not be `fileStat`? as stat is returning an fs.Stat object?

Reply:
Nope :) It would be to easy... for some reason it's called stats in the code and documentation.
Probably plural since every field in it is considered a "stat".

https://nodejs.org/api/fs.html#class-fsstats
https://nodejs.org/api/fs.html#fsstatpath-options-callback

Reply:
I see your point there’s some inconsistencies… Class is Stats and it contains n properties whilst the function is stat that return one instance of Class Stats.

Reply:
stat is the verb, while stats is the result type.